### ISSUE TITLE
Add option to ignore .nomedia files

### ISSUE THREAD

**Issue by adrian-bl** [vanilla-music/vanilla#568]

**Comment by adrian-bl** [comment]
Closing this as obsoleted as we now have an option to explicitly include folders

**Comment by PeteSapai** [comment]
Hi there,

I've emailed about this, but haven't gotten a reply, so I thought I'd comment here.

Explicitly including a folder does not work for me. I put a .nomedia file in my music folder on the sdcard. I go to 'Settings->Media Library->Indexed Directories->Edit', then browse to /storage/sd-card/, long-press the Music folder and select Include, it is added to 'indexed directories' (and is the only directory enabled there, but nothing shows up in Artists, Albums, etc. When I use Files and browse to the Music dir, long-press and select Play or Play Next, etc. it says "0 tracks playing". If I use the 'scan full filesystem' option, it finishes immediately. If I enable another folder, it will scan for a while, but finds only 2 hangouts files. Files do show up in the browser, and playing (or enqueuing) single files does work, but that's it.

tl;dr: there is no way to play folders with a .nomedia file present

**Comment by adrian-bl** [comment]
> tl;dr: there is no way to play folders with a .nomedia file present

Because we ignore folders with .nomedia - Adding a folder to the scanner just tells Vanilla to look at it, it doesn't 'forcefully' include it.

But you can just include a subfolder which does NOT have .nomedia.

Eg:

/sdcard/Music/.nomedia
/sdcard/Music/Files

Including the 'Files' folder with such a layout would work while stopping the builtin android media scanner from indexing /sdcard/Music/...

**Comment by PeteSapai** [comment]
But  then this issue/reqeust (option to ignore .nomedia files) isn't obsolete, is it? There's "only" a workaround.

Oh well, your solution works (I still think it's not ideal, but it works nonetheless :)): I put a .nomedia file in the root of the sdcard, added /sdcard/music, started a full scan and it now recognises the music dir.
Thanks!

**Comment by bbaserdem** [comment]
This is an old issue, but this would be a very useful feature still. My usecase (and I see that it is common online) is to put a nomedia file in your music folder so it hides album covers showing up on your pictures galery, and hide your music library from other apps on your phone like VLC. Then you can specify to your music app where your music library is. With 10000 songs, it is not viable for me to pick each subfolder in my library to add to the scan path.

**Comment by xlotlu** [comment]
@adrian-bl please reopen this. My use-case is exactly the same with @bbaserdem's: keeping album covers away from the gallery.

I would, in fact, consider the cureent implementation buggy: specifically selecting a folder should ignore the existence of a `.nomedia` in said folder, and only apply to its descendants. If you agree with this logic, an "ignore .nomedia" option would be redundant I guess.
